backend/app/api/upload.py

```diff
@@ -40,6 +45,51 @@ def post(self):
                 image.save(path, image.format)
             #filename = photos.save(request.files['photos'])
             return "Success", 201
+        elif 'interactions' in request.files:
+            print "In interactions!!"
+            print request.form
+            interaction_id = request.form.get('interaction_id')
+            dir_path = os.path.join(dest, interactions_sub, interaction_id)
+            print "Path is: " + dir_path
+            if not os.path.exists(dir_path):
+                print "making the dir!"
+                os.makedirs(dir_path)
+            path = os.path.join(dir_path, request.files['interactions'].filename)
+            if os.path.exists(path):
+                os.remove(path)
+            print request.files
+            filename = interactions_upload_set.save(request.files.get('interactions'), interaction_id)
+            print filename
+            return "Success", 201
         else:
             msg = 'No file included in upload request'
             abort(400, message=msg)
+        
+class InteractionFiles(Resource):
+    def get(self, id):
+        filenames = self.getFileList(id);
+        response = {"filenames": filenames}
+        return response, 200
+    
+    # This will delete any files that are not in the posted list
+    def post(self, id):
+        path = os.path.join(dest, interactions_sub, id)
+        existing_files = self.getFileList(id)
+        posted_files = list(request.get_json())
+        files_to_delete = [x for x in existing_files if x not in posted_files]
+        for file in files_to_delete:
+            file_path = os.path.join(path, file)
+            if not os.path.exists(file_path):
+                msg = "File not found"
+                abort(401, message=msg)
+            else:
+                os.remove(file_path)
+        return 204
+
+    def getFileList(self, id):
```

Comment:
Not a big deal but using camel case isn't consistent with how we've been naming methods in this project for python.
Maybe we should look into adding a linting step as part of our TravisCI? Made #151 to track this decision

Reply:
Oh! I understand now, naming decision was made so maintain consistency in the typescript code. An interesting problem. Well, maybe static checkers that fail the build wouldn't be such a good idea after all